src/qt/walletmodel.cpp

```diff
@@ -756,3 +779,296 @@ bool WalletModel::getDefaultWalletRbf() const
 {
     return fWalletRbf;
 }
+
+bool WalletModel::nameAvailable(const QString &name)
+{
+    JSONRPCRequest jsonRequest;
+    UniValue params (UniValue::VOBJ);
+    UniValue res, array, isExpired;
+
+    const std::string strName = name.toStdString();
+    params.push_back (Pair("name", strName));
+    jsonRequest.strMethod = "name_show";
+    jsonRequest.params = params;
+    jsonRequest.fHelp = false;
+
+    try {
+        res = tableRPC.execute(jsonRequest);
+
+    } catch (const UniValue& e) {
+        return true;
+    }
+
+    isExpired = find_value( res, "expired");
+    if(isExpired.get_bool())
+      return true;
+
+    return false;
+}
+
+NameNewReturn WalletModel::nameNew(const QString &name)
+{
+    std::string strName = name.toStdString ();
+    std::string data;
+
+    JSONRPCRequest jsonRequest;
+    UniValue params(UniValue::VOBJ);
+    std::vector<UniValue> values;
+    UniValue res, txid, rand;
```

Comment:
Please move the declaration of `values`, `res`, `txid` and `rand` to below where you actually use the variables.